homeassistant/components/samsungtv/bridge.py

```diff
@@ -84,19 +86,23 @@ async def async_get_device_info(
     if bridge and bridge.port:
         return bridge.port, bridge.method, await bridge.async_device_info()
 
-    # Try websocket ports
-    for port in WEBSOCKET_PORTS:
-        bridge = SamsungTVBridge.get_bridge(hass, METHOD_WEBSOCKET, host, port)
-        if info := await bridge.async_device_info():
-            return port, METHOD_WEBSOCKET, info
+    # Try non-ssl websocket ports
+    bridge = SamsungTVBridge.get_bridge(
+        hass, METHOD_WEBSOCKET, host, WEBSOCKET_NO_SSL_PORT
+    )
+    if info := await bridge.async_device_info():
+        encrypted_bridge = SamsungTVEncryptedBridge(
+            hass, METHOD_ENCRYPTED_WEBSOCKET, host, ENCRYPTED_WEBSOCKET_PORT
+        )
+        if await encrypted_bridge.async_can_connect():
```

Comment:
async_try_connect on the encrypted broke the legacy tv discovery so I added `async_can_connect` but it needs to be moved into the upstream lib